### ISSUE TITLE
BERT and Hyperbolic embedding for ICD coding task

### ISSUE THREAD

**Issue by derejeweyessaa** [derejeweyessaa/HyperBERT-#1]
Hyper BERT Model: Intelligent ICD Code Assignment for Clinical Narratives Using BERT and Hyperbolic space for Ethiopia National Classification of Diseases
Authors:1. Dereje Gizaw (PhD candidate at ASTU; derejeweyessaa@gmail.com; 2.Dr. Beranu Beyene ,Associate Professor and President of Ethiopia Cyber Security Association, email: berhanebeyen@gmail.com; 3.Dr. Teklu Urgessa , Associate Professor and Dean of School of Electrical Engineering and Computing at ASTU , email: tekluurgessa2009@gmail.com
                                                                  May, 2024
Abstract
Automatic International classification of disease (ICD) coding is important because manually assigning codes is expensive, time-consuming, and error-prone. The topic is addressing ICD code assigning to clinical narratives text - a vital problem in the healthcare sector that otherwise is manually processed. The focus of this research study is to design mechanisms on how to embed the code hierarchal tree structure of the ICD code on the task  automatic ICD code assignment clinical text, namely: (1) how to introduce diagnosis code hierarchy as auxiliary information knowledge in the training process BERT approach? (2) How transformer-based models like BERT can be applied as practice-oriented, interpretable AI  considering Ethiopia's national classification of disease  (NCOD)  and Medical Information Mart in Intensive care unit (MIMIC)  data set? The research design approach applied  is qualitative with research strategies of experimental study and case study. The experiment result gained show that the new model architecture called HyperBERT  increase macro average AUC 0.003 and 0.02 of F1 score , and demonstrated the interpretability of the model using code-wise label attention mechanism using BERT visualization tool, unlike the state-of-the-art models. The  discussion section analyzed the effectiveness of our model as code-aware document representations . Finally the researchers concluded that the model is viable and accurate as code-aware document representations by incorporating the Poincare distance as the loss function and using Riemannian adaptive optimization techniques with mathematical proof description, and recommend the applicability of the new model as way forward to integrate with healthcare systems in Ethiopia. 
Key Words— ICD Coding; Explainable AI, Code Hierarchical; Neural Network; BERT; Hyperbolic embedding.
